File: backend/app/constants/tour_transport.py
"""Suggested transport modes for tour packages."""

from __future__ import annotations

BUS = "bus"
CAR = "car"
TRAIN = "train"
FLIGHT = "flight"
VAN = "van"
TEMPO = "tempo"
CRUISE = "cruise"
BOAT = "boat"
BIKE = "bike"
OTHER = "other"

SUGGESTED_TRANSPORT_TYPES: tuple[str, ...] = (
    BUS,
    CAR,
    TRAIN,
    FLIGHT,
    VAN,
    TEMPO,
    CRUISE,
    BOAT,
    BIKE,
)

TRANSPORT_TYPE_LABELS: dict[str, str] = {
    BUS: "Bus",
    CAR: "Car",
    TRAIN: "Train",
    FLIGHT: "Flight",
    VAN: "Van",
    TEMPO: "Tempo Traveller",
    CRUISE: "Cruise",
    BOAT: "Boat",
    BIKE: "Bike / Scooter",
}


def transport_type_label(value: str | None) -> str | None:
    if not value:
        return None
    key = str(value).strip().lower()
    return TRANSPORT_TYPE_LABELS.get(key, str(value).strip())


def normalize_transport_type(value) -> str | None:
    if value is None:
        return None
    text = str(value).strip()
    if not text:
        return None
    if len(text) > 60:
        raise ValueError("transport_type must be at most 60 characters")
    lowered = text.lower()
    if lowered in TRANSPORT_TYPE_LABELS:
        return lowered
    return text

File: frontend/src/utils/tourTransport.js
/** Tour package transport modes — keep labels aligned with backend `tour_transport` constants. */

export const TOUR_TRANSPORT_OTHER = 'other';

export const TOUR_TRANSPORT_OPTIONS = [
  { value: 'bus', label: 'Bus' },
  { value: 'car', label: 'Car' },
  { value: 'train', label: 'Train' },
  { value: 'flight', label: 'Flight' },
  { value: 'van', label: 'Van' },
  { value: 'tempo', label: 'Tempo Traveller' },
  { value: 'cruise', label: 'Cruise' },
  { value: 'boat', label: 'Boat' },
  { value: 'bike', label: 'Bike / Scooter' },
  { value: TOUR_TRANSPORT_OTHER, label: 'Other' },
];

const LABEL_BY_VALUE = Object.fromEntries(
  TOUR_TRANSPORT_OPTIONS.map((option) => [option.value, option.label]),
);

export function tourTransportLabel(value) {
  if (!value) return '';
  const key = String(value).toLowerCase();
  return LABEL_BY_VALUE[key] || String(value);
}

export function resolveTourTransportPayload(transportType, transportOther) {
  const selected = (transportType || '').trim();
  if (!selected) return null;
  if (selected === TOUR_TRANSPORT_OTHER) {
    return (transportOther || '').trim() || null;
  }
  return selected;
}

export function splitTourTransportValue(value) {
  if (!value) {
    return { transport_type: '', transport_type_other: '' };
  }
  const key = String(value).toLowerCase();
  if (LABEL_BY_VALUE[key]) {
    return { transport_type: key, transport_type_other: '' };
  }
  return { transport_type: TOUR_TRANSPORT_OTHER, transport_type_other: String(value) };
}
